plugins/node/opentelemetry-instrumentation-aws-lambda/src/aws-lambda.ts

```diff
@@ -0,0 +1,216 @@
+ * limitations under the License.
+ */
+
+import * as path from 'path';
+
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import {
+  diag,
+  Span,
+  SpanKind,
+  SpanStatusCode,
+  TracerProvider,
+} from '@opentelemetry/api';
+import { CLOUD_RESOURCE } from '@opentelemetry/resources';
+import { FaasAttribute } from '@opentelemetry/semantic-conventions';
+
+import { Callback, Context, Handler } from 'aws-lambda';
+
+import { LambdaModule } from './types';
+import { VERSION } from './version';
+import { NodeTracerProvider } from '@opentelemetry/node';
+
+export class AwsLambdaInstrumentation extends InstrumentationBase {
+  private _tracerProvider: TracerProvider | undefined;
+
+  constructor() {
+    super('@opentelemetry/instrumentation-awslambda', VERSION);
+  }
+
+  init() {
+    // _HANDLER and LAMBDA_TASK_ROOT are always defined in Lambda.
+    const taskRoot = process.env.LAMBDA_TASK_ROOT!;
+    const handlerDef = process.env._HANDLER!;
+
+    const handler = path.basename(handlerDef);
+    const moduleRoot = handlerDef.substr(0, handlerDef.length - handler.length);
+
+    const [module, functionName] = handler.split('.', 2);
+
+    // Lambda loads user function using an absolute path.
+    let filename = path.resolve(taskRoot, moduleRoot, module);
+    if (!filename.endsWith('.js')) {
+      // Patching infrastructure currently requires a filename when requiring with an absolute path.
+      filename += '.js';
+    }
+
+    return [
+      new InstrumentationNodeModuleDefinition(
+        // NB: The patching infrastructure seems to match names backwards, this must be the filename, while
+        // InstrumentationNodeModuleFile must be the module name.
+        filename,
+        ['*'],
+        undefined,
+        undefined,
+        [
+          new InstrumentationNodeModuleFile(
+            module,
+            ['*'],
+            (moduleExports: LambdaModule) => {
+              diag.debug('Applying patch for lambda handler');
+              if (isWrapped(moduleExports[functionName])) {
+                this._unwrap(moduleExports, functionName);
+              }
+              this._wrap(moduleExports, functionName, this._getHandler());
+              return moduleExports;
+            },
+            (moduleExports?: LambdaModule) => {
+              if (moduleExports == undefined) return;
+              diag.debug('Removing patch for lambda handler');
+              this._unwrap(moduleExports, functionName);
+            }
+          ),
+        ]
+      ),
+    ];
+  }
+
+  private _getHandler() {
+    return (original: Handler) => {
+      return this._getPatchHandler(original);
+    };
+  }
+
+  private _getPatchHandler(original: Handler) {
+    diag.debug('patch handler function');
+    const plugin = this;
+
+    return function patchedHandler(
+      this: {},
```

Comment:
Maybe use `this: never` as you just forward. This protects from accidentally using it.